homeassistant/components/guardian/__init__.py

```diff
@@ -213,19 +214,15 @@ class GuardianEntity(CoordinatorEntity):
     """Define a base Guardian entity."""
 
     def __init__(  # pylint: disable=super-init-not-called
-        self,
-        entry: ConfigEntry,
-        kind: str,
-        name: str,
-        device_class: str | None,
-        icon: str | None,
+        self, entry: ConfigEntry, description: EntityDescription
     ) -> None:
         """Initialize."""
-        self._attr_device_class = device_class
+        self._attr_device_class = description.device_class
         self._attr_device_info = {"manufacturer": "Elexa"}
         self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: "Data provided by Elexa"}
-        self._attr_icon = icon
-        self._attr_name = name
+        self._attr_icon = description.icon
+        self._attr_name = description.name
+        self._description = description
```

Comment:
Assign `description` to `self.entity_description` instead. That way `_attr_icon`, `_attr_name`, `_attr_device_class`, and so on are automatically populated. No need to do it manually.

The assignment happens in the `Entity`, `SensorEntity` base classes.